kartothek/io_components/read.py

```diff
@@ -75,7 +75,7 @@ def dispatch_metapartitions_from_factory(
 
         # Group the resulting MetaParitions by partition keys or a subset of those keys
         merged_partitions = base_df.groupby(
-            by=list(dispatch_by), sort=False, as_index=False
+            by=list(dispatch_by), sort=True, as_index=False
```

Comment:
when is this used? I tried reverting to `sort=False` and the test below was still passing

Reply:
Also note, `dispatch_by` is a string here

Reply:
> when is this used? I tried reverting to sort=False and the test below was still passing

Should add a test case then. I rely on having the partitions sorted when setting the divisions.

- [ ] Add test case for sorted/unsorted partitions

Reply:
> Also note, dispatch_by is a string here

Should be normalised

- [ ] Add normalisation for dask_index_on